PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightScalaPageTwo.java

```diff
@@ -51,6 +63,56 @@ public void configureJavaProject(String newProjectCompliance, IProgressMonitor m
 		}
 	}
 	
+	private void addMoreSourcetoClassPath() throws JavaModelException {
+		if (parent == null) { 
+			parent = (HDInsightsScalaProjectWizard) this.getWizard();
+		}
+		
+		if (parent.useMaven) {
+			IProject project = this.getJavaProject().getProject();
+			final IFolder sourceRootFolder = project.getFolder("src");
+			if (!sourceRootFolder.exists()) {
+				try {
+					sourceRootFolder.create(false, true, null);
+				} catch (CoreException e) {
+					e.printStackTrace();
+				}
+			}
+
+			final IFolder mainFolder = sourceRootFolder.getFolder("main");
+			if (!mainFolder.exists()) {
+				try {
+					mainFolder.create(false, true, null);
+				} catch (CoreException e) {
+					e.printStackTrace();
+				}
+			}
+			
+			final IFolder scalaFolder = mainFolder.getFolder("scala");
+			if (!scalaFolder.exists()) {
+				try {
+					scalaFolder.create(false, true, null);
+				} catch (CoreException e) {
+					e.printStackTrace();
+				}
+			}
+			
+			IJavaProject javaProject = this.getJavaProject();
+			
+			IClasspathEntry[] entries = javaProject.getRawClasspath();
+	
+			IClasspathEntry[] newEntries = new IClasspathEntry[entries.length];
+			
+			IClasspathEntry scalaSrcFolder = JavaCore.newSourceEntry(scalaFolder.getFullPath());
+	
+			System.arraycopy(entries, 0, newEntries, 0, entries.length);
```

Comment:
Can `ArrayList.clone()` work for this? The arraycopy and the following array operation looks dangerous.

Reply:
Rewrote this part

---
In reply to: [136774411](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136774411) [](ancestors = 136774411)

Reply:
Need to remove one entry from the array. So arraycopy is needed.

---
In reply to: [138296662](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r138296662) [](ancestors = 138296662,136774411)